extensions/jms/src/main/java/org/mule/extensions/jms/api/source/JmsListener.java

```diff
@@ -172,23 +178,35 @@ public void onStop() {
     if (LOGGER.isDebugEnabled()) {
       LOGGER.debug("Stopping JMSSubscriber source");
     }
+
+    jmsLock.unlock();
+
+    try {
+      if (ackMode.equals(AUTO) || ackMode.equals(DUPS_OK) || ackMode.equals(MANUAL)) {
+        session.get().recover();
```

Comment:
the old transport had the concept of how many consumers to have on each destination. When the source was stopped, the stop logic had to be done on each of those consumers. Where is that concept in this connector? Shouldn't the ACK logic be done for each of them?